src/nupic/engine/Network.cpp

```diff
@@ -670,7 +672,12 @@ void Network::load(std::istream &f) {
   f.ignore(1);
 
   // Post Load operations
-  initialize();   //  re-initialize everything
```

Comment:
why calling initialize in deserialization does not work?

Reply:
> why calling initialize in deserialization does not work?

initialize() in Network ends up calling initialize in the Region Implementations which re-instantiates the algorithm so all state is lost.  So we cannot call initialize when deserializing.